### ISSUE TITLE
Activating a tab based on the URL hash should also set focus to the corresponding tabpanel

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8483]
### Is your proposal related to a problem?

In the Wagtail admin interface, for tabbed UIs, we create links to specific tabs like: `http://mywagtail.example.com/admin/pages/60/edit/#tab-settings`. When a user loads the page with a link containing a tab URL hash (`#tab-settings`), we then activate the corresponding tab. But we keep focus at the top of the document, which is unexpected.

### Describe the solution you'd like

It would be nice for us to also place keyboard focus directly on the corresponding tabpanel – matching what we do with our skip link, and what people generally seem to expect when linking to a given section of a page.

Here is a screenshot of the result:

![option-3-onload-tabpanel](https://user-images.githubusercontent.com/877585/166693122-cfd04d74-ba79-4950-b3c7-4ff5c416904b.png)

It’s unclear to me whether it makes sense for the tabpanel to have a visible focus as shown here or not.

### Describe alternatives you've considered

This issue is based on a [slack thread](https://wagtailcms.slack.com/archives/CB7L6L5S6/p1651584317720309) where I proposed three options in total:

1. Top of the document, same as the link without the tab identifier.
2. On the corresponding tab within the tablist, with the first focusable item within the active tabpanel as the next item in the keyboard tab order
3. On the corresponding tabpanel, with the first focusable item within the tabpanel as the next item in the keyboard tab order

### Additional context

For other aspects of our tabs implementation, we follow [ARIA authoring practices for tabs](https://w3c.github.io/aria-practices/#tabpanel). I’ve researched guidance on tabs supporting deep linking and didn’t find much.

**Comment by lb-** [comment]
Note that this line was probably intended to do the same thing (for create only) and very likely no longer works
https://github.com/wagtail/wagtail/blob/1297b930e54ec3d7255a5f56e3b5cb85724a4439/wagtail/admin/templates/wagtailadmin/pages/create.html#L94

This is due to the focus being called on DOMContentLoaded but the timing of the tabs and the active section may not align.

**Comment by thibaudcolas** [comment]
@lb- this looks more like a hacky "autofocus first field", which I think we should remove.

**Comment by lb-** [comment]
@salty-ivy  here is a JS only one - you may be keen on investigating.

**Comment by salty-ivy** [comment]
Would look into it

**Comment by salty-ivy** [comment]
@lb-  How about  achieving this via stimulus controller?

**Comment by lb-** [comment]
@salty-ivy yeah. I agree - but the tabs are not yet migrated to Stimulus, I was thinking that this other ticket may make sense for that migration piece. 

https://github.com/wagtail/wagtail/issues/8408

Let me know if you're keen to pick it up and I'll either raise a new issue or re-word 8408 to include Stimulus migration info.

**Comment by salty-ivy** [comment]
> @salty-ivy yeah. I agree - but the tabs are not yet migrated to Stimulus, I was thinking that this other ticket may make sense for that migration piece.
> 
> #8408
> 
> Let me know if you're keen to pick it up and I'll either raise a new issue or re-word 8408 to include Stimulus migration info.

Sure ! why not, although little busy for few days, definitely be back to it in a week or so and I would be completing https://github.com/wagtail/wagtail/issues/10133 first.
